src/cargo/ops/cargo_compile.rs

```diff
@@ -110,22 +111,33 @@ impl Packages {
     }
 
     pub fn to_package_id_specs(&self, ws: &Workspace<'_>) -> CargoResult<Vec<PackageIdSpec>> {
-        let specs = match *self {
+        let specs = match self {
             Packages::All => ws
                 .members()
                 .map(Package::package_id)
                 .map(PackageIdSpec::from_package_id)
                 .collect(),
-            Packages::OptOut(ref opt_out) => ws
-                .members()
-                .map(Package::package_id)
-                .map(PackageIdSpec::from_package_id)
-                .filter(|p| opt_out.iter().position(|x| *x == p.name()).is_none())
-                .collect(),
-            Packages::Packages(ref packages) if packages.is_empty() => {
+            Packages::OptOut(opt_out) => {
+                let mut opt_out: HashSet<&String> = HashSet::from_iter(opt_out);
```

Comment:
Could this perhaps use `BTreeSet` or similar to have deterministic output? (should make it a bit easier to test)

Reply:
Sure!